---
title: "Analysis 2: Filtering Analysis"
author: "Joe Gunn"
date: "2023-04-28"
output: html_document
---

# Project: Effects of admixture on fitness in Neosho Bass populations 
<font size="+1">We assessed the effect of admixture on fitness in two stream populations within the native range of the Neosho Bass (<i>M. velox</i>) which are known to have extensively hybridized with Smallmouth Bass (<i>Micropterus dolomieu</i>). Specifically, we used 14 microsatellite loci in a Bayesian analysis of population structure to estimate proportions of interspecific ancestry in individuals collected from Big Sugar Creek and the Elk River in southwestern Missouri (Central Interior Highlands ecoregion (CIH), North America). We used ancestry inference to identify fish as "Pure Neosho Bass", "Pure Smallmouth Bass", or "Admixed". For each group, we measured age and total length and projected individual growth using the standard parameterization of the von Berlanffy growth model, comparing average theoretical maximum length among groups. Finally, we used calculated a body condition as a proxy of fitness and generated heterozygosity-fitness correlations of body condition across the global dataset, within stream populations, and within ancestry groups. We ultimately sought to understand the short-term genetic consequences of admixture for Neosho Bass populations in order to better inform management and long-term viability of distinct, economically and ecologically important sportfish species in the CIH.</font>

## Specific Aim: Data preparation, enumeration, and filtering
For this aim, we clean, filter, and summarize the full genotype data (14 microsatellite loci) and phenotypic data (consensus age and total length) for 135 raw samples, which comprised fish collected from two streams (Big Sugar Creek and Elk River) within the recognized native range of the Neosho Bass (<i>M. velox</i>; NB) and a reference set of samples representing Smallmouth Bass (<i>M. dolomieu</i>; SMB) collected from streams in the White River drainage in the Smallmouth Bass native range. We did not collect phenotypic data on the Smallmouth Bass reference samples.

## Phases of Analysis
### Phase 1: Data read-in and filtering
### Phase 2: Data summarization and enumeration

## Libraries needed for analysis
```{r setup}
library(readxl)
library(tidyverse)
library(cowplot)
```

## PHASE 1: DATA READ-IN AND PREPARATION

### STEP 1: Read in metadata for samples (all fish collected from the Neosho Bass range) and references (all fish collected from the Smallmouth Bass range)
In this step, we read in previously curated metadata for fish collected from two streams within the Neosho Bass native range and for a reference set of fish collected from three streams within the Smallmouth Bass native range (`../raw_data/metadata.xlsx`). We manipulate, clean and enumerate the dataset, and fully summarize all samples in PHASE 2 below.

#### 1a. Read in and clean sample metadata; run the Rmd chunk below.
In this step, we read in the full metadata for all fish included in the study. Data include: 

   1. "sample_id": a unique, alphanumeric ID for each sample
   2. "river_code": the alphabetical prefix for each sample id, where "FBS" stands for "Fitness, Big Sugar Creek" and "FER" stands for "Fitness, Elk River"
   3. "sample_number": the numerical prefix for each sample (01, 02, 03,..., 71)
   4. "structure_number": the alphanumerical ID for each individual in the proper format for analysis with the program STRUCTURE
   5. "easting": easting UTM (ZONE 15S) measurement (GIS)
   6. "northing": northing UTM (ZONE 15S) measurement (GIS)
   7. "dist_from_df": distance (m) from Deep Ford Access (DFA) at which each sample was collected
   8. "range_id": species native range from which each individual was collected
   9. "river": river or stream frow which each individual was collected 
   10. "population": designation for whether the sample was included in growth analyses ("sample") or was used solely as a reference for ancestry analysis ("reference")

##### Read in and clean sample metadata:
```{r}
# Read in sample metadata
metadata <- read_excel("../raw_data/metadata.xlsx") 

# Convert characters to factors
metadata <- metadata %>%
  mutate(sample_id = factor(sample_id)) %>%
  mutate(structure_number = factor(structure_number)) %>%
  mutate(river_code = factor(river_code)) %>%
  mutate(sample_number = as.numeric(sample_number)) %>%
  mutate(easting = as.numeric(easting)) %>%
  mutate(northing = as.numeric(northing)) %>%
  mutate(dist_from_df = as.numeric(dist_from_df)) %>%
  mutate(range_id = factor(range_id)) %>%
  mutate(river = factor(river)) %>%
  mutate(population = factor(population))
```

#### 1b. Enumerate full metadata with samples and reference; run the Rmd chunk below.
In this step, we conduct a preliminary data summary to see how many samples are in the raw metadata. We compare all metadata with raw data files (genotypes and phenotypes) in STEP 2b.1 and STEP 3b.1 below.

##### Summarize metadata:
```{r}
# Get full sample number 
metadata %>% 
  count()

# Get sample size of sample and reference sets
metadata %>%
  group_by(population) %>%
  count()
  
# Get sample size by stream of origin
metadata %>%
  group_by(population, 
           river) %>%
  count()
```

<b>Data summary:</b> <br>

<b><i>N</i><sub>total</sub> = 136 </b><br>

<b><i>N</i><sub>sample_set</sub> = 116 </b><br>
<i>N</i><sub>sample_elk</sub> = 70 <br>
<i>N</i><sub>sample_bigsugar</sub> = 46 <br>

<b><i>N</i><sub>reference_set</sub> = 20 </b><br>
<i>N</i><sub>reference_white</sub> = 2 <br>
<i>N</i><sub>reference_tablerock</sub> = 8 <br>
<i>N</i><sub>reference_crooked</sub> = 10 <br>

### STEP 2: Read in genotype data for all samples and references (one single genotype file)
In this step, we read in genotype data for all fish included in the study, including samples (collected from the Neosho Bass native range) and references (collected from the Smallmouth Bass native range). These genotype data were derived from multiplex PCR (see `../project_info/msat_primer_info/msat_primer_info.docx` for microsatellite primer oligonucleotide flanking sequences and corresponding metadata, including literature references) followed by and fragment analysis and subsequent allele scoring in GeneMarker (Soft Genetics). We manipulate, clean and enumerate the dataset, and fully summarize all samples in PHASE 2 below.

#### 2a. Read in and clean genotype data; run the Rmd chunk below.
In this step, we read in genotype data for all fish included in the study. Data include: 

   1. "sample_id": same as in metadata
   2. "structure_number": same as in metadata 
   3. two columns for each microsatellite locus, one column per allele (see description under STEP 2 above for info)

##### Read in and clean genotype data:
```{r}
# Read in genotype data
genotype_data <- read_excel("../raw_data/genotype_data.xlsx") 

# Convert characters to factors
genotype_data <- genotype_data %>%
  mutate(sample_id = factor(sample_id)) %>%
  mutate(structure_number = factor(structure_number))
```

#### 2b. Enumerate full genotype data; run the Rmd chunk below.
In this step, we determine the total number of samples in the dataset.

##### Summarize genotype data:
```{r}
# Get full sample number (N = 136)
genotype_data %>% 
  count()
```

##### 2b.1. Compare metadata and genotype data to check for sample number discrepancies; run the Rmd chunk below.
In this step, we compare the metadata and genotype data to see if there are any discrepancies in samples between datasets (samples are present in one, but not the other).

##### Resolve sample number discrepancy:
```{r}
# Samples in metadata but not in genotype data
metadata %>% 
  filter(!metadata$sample_id %in% genotype_data$sample_id)

# Samples in genotype data but not in metadata
genotype_data %>% 
  filter(!genotype_data$sample_id %in% metadata$sample_id)
```
We found no discrepancies in the data.

### STEP 3: Read in phenotype data data for samples
In this step, we read in phenotype data for all fish included in the study. Data include: 

   1. "sample_id": same as in metadata and genotype data (followed by "structure_number"; see Analysis 3, Phase 1, step 1e)
   2. "sex": male or female, determined by internal gonad examination, 
   3. "tl_alive": total length of the fish immediately upon capture by rod-and-reel, as measured from the tip of the lower mandible to the posterior tip of the caudal fin squeezed together
   4. "tl_dead": total length of the fish upon sample processeing (after approximately one month frozen)
   5. "mass_dead": mass of the fish in grams at time of sample processing
   6. "meas_ints": initials of the person doing the phenotypic measuring, measuring, or assessment, for later analysis of observer/measurement bias
   7. "eddie_age": age estimate based on sagittal otolith annuli, performed by Eddie Sterling
   8. "joe_age": age estimate performed by Joe Gunn
   9. "michael_age": age estimate performed by Michael Moore 
   10. "consensus_age": ultimate age estimate based on consensus of three agers, Eddie, Joe, and Michael
   11. "sl": standard length of the fish, as measured from the tip of the mandible to the hyperal plate, just anterior to the caudal fin
   12. "bd": body depth, as measured at the thickest part of the fish
   13. "hl": head length, as measured from the tip of the lower mandible to the posterior end of the gill plate
   14. "ol": orbital length, the diameter of the eye ball
   15. "sd_ray": the number of soft dorsal fin rays. 

Additionally, we did not collect phenotype data for the reference set (Smallmouth Bass), as these samples were used only as a reference for genetic ancestry inference in STRUCTURE in subsequent analyses. We manipulate, clean and enumerate the dataset, and fully summarize all samples in PHASE 2 below.

#### 3a. Read in and clean the phenotype dataset; run the Rmd chunk below.

##### Read in and clean phenotype data:
```{r}
# Read in phenotype data
phenotype_data <- read_excel("../raw_data/phenotype_data.xlsx") 

# Convert characters to factors
phenotype_data <- phenotype_data %>%
  mutate(sample_id = factor(sample_id)) %>%
  mutate(structure_number = factor(structure_number)) %>%
  mutate(sex = factor(sex)) %>%
  mutate(tl_alive = as.numeric(tl_alive)) %>%
  mutate(tl_dead = as.numeric(tl_dead)) %>%
  mutate(mass_dead = as.numeric(mass_dead)) %>%
  mutate(meas_ints = factor(meas_ints)) %>%
  mutate(eddie_age = as.numeric(eddie_age)) %>%
  mutate(joe_age = as.numeric(joe_age)) %>%
  mutate(michael_age = as.numeric(michael_age)) %>%
  mutate(consensus_age = as.numeric(consensus_age)) %>%
  mutate(sl = as.numeric(sl)) %>%
  mutate(bd = as.numeric(bd)) %>%
  mutate(hl = as.numeric(hl)) %>%
  mutate(ol = as.numeric(ol)) %>%
  mutate(sd_ray = as.numeric(sd_ray))
```

#### 3b. Enumerate full phenotype data; run the Rmd chunk below.
In this step, we determine the total number of samples in the phenotype dataset.

##### Summarize phenotype data:
```{r}
# Get sample set number
phenotype_data %>% 
  count()
```

There was a discrepancy in the number of sample ids present in the metadata for the sample set (<i>N</i> = 136) and the number present in the phenotype data (<i>N</i> = 139). We therefore conducted a direct comparison of the two columns to see where they differed and to identify any sample ids without corresponding data in both datasets.

##### 3b.1. Compare metadata and phenotype data to resolve sample number discrepancy; run the Rmd chunk below.
In this step, we compare the metadata and phenotype data to see if there are any discrepancies in samples between datasets (samples are present in one, but not the other).

##### Resolve sample number discrepancy:
```{r}
# Samples in metadata but not in phenotype data
metadata %>% 
  filter(!metadata$sample_id %in% phenotype_data$sample_id)

# Samples in genotype data but not in metadata
phenotype_data %>% 
  filter(!phenotype_data$sample_id %in% metadata$sample_id)
```

We found that the phenotype data file contained three samples (FBS47, FBS48, FBS49) that were not represented in the metadata after previous filtering (see STEP 2 above). The three unmatched samples were collected at a later time compared to all other samples, and we did not take genotype data on these fish. Samples FBS47, FBS48, and FBS49 were therefore not considered part of the final dataset and were removed from downstream analyses (see Step 3b.2 below).

##### 3b.2. Remove unmatched sample id; run the Rmd chunk below.

##### Resolve sample number discrepancy:
```{r}
# Remove sample FER71 from the metadata set
phenotype_data <- phenotype_data %>%
  filter(sample_id != "FBS47") %>%
  filter(sample_id != "FBS48") %>%
  filter(sample_id != "FBS49")
```

### STEP 4: Save all datasets for downstream analyses
In this step, we save all filtered datasets for final merging (see STEP 5 below) and summary (see PHASE 2 below).

#### 4a. Save all datasets
```{r}
# Save metadata
save(metadata, file = "data/processed_raw_data/metadata.Rda")

# Save genotype data
save(genotype_data, file = "data/processed_raw_data/genotype_data.Rda")

# Save phenotype data
save(phenotype_data, file = "data/processed_raw_data/phenotype_data.Rda")
```

### STEP 5: Merge all datasets (metadata, phenotype_data, genotype_data)
In this step, we create three working datasets: 1) metadata and genotype data (`data/processed_raw_data/full_genotype_data.Rda`); 2) metadata and phenotype_data (`data/processed_raw_data/full_phenotype_data.Rda`); and 3) full data, including metadata, genotype data, and full data (`data/processed_raw_data/full_data.Rda`). These files will be used in subsequent analyses and subsetted where necessary.

#### 5a. Merge data sets; run the Rmd chunk below.

##### Merge and create full metadata and full phenotype data
```{r}
# Load in metadata
load("data/processed_raw_data/metadata.Rda")

# Load in genotype_data
load("data/processed_raw_data/genotype_data.Rda")

# Load in phenotype_data
load("data/processed_raw_data/phenotype_data.Rda")

## Full genotype data

# Get full genotype dataset
full_genotype_data <- merge(metadata, 
                   genotype_data,
                   by = c("sample_id",
                          "structure_number"))

## Full phenotype data

# Get full phenotype dataset
full_phenotype_data <- merge(metadata, 
                   phenotype_data,
                   by = c("sample_id",
                          "structure_number"))

# Merge all data
full_data <- merge(full_genotype_data, 
                   full_phenotype_data,
                   by = c("sample_id",
                          "structure_number",
                          "river_code",
                          "sample_number",
                          "easting",
                          "northing",
                          "dist_from_df",
                          "range_id",
                          "river",
                          "population"))
```

Here, we found that one individual in the Elk River, Neosho Bass range (FER71) was missing data for tl_alive (total length while alive). We were therefore not able to assign this fish to a size category. However, we did have consensus age (age = 2) and tl_dead (total length while dead, 224.0mm). To maximize sample size, we retained this sample by imputing tl_alive as the expected value at 224.0mm for age-2 fish based on a mixed-effect linear model (with stream as a random effect to account for any variation between stream ecologies), where x = tl_dead and y = tl_alive. 

#### 5b. Impute total length while alive (tl_alive) for sample FER71; run the Rmd chunk below:
In this step, we filter the dataset to only age-2 fish, and we examine the linear relationship between total length while dead (tl_dead) and total length while alive (tl_alive) by testing for a significant correlation between these variables. We use the resulting linear model to impute the total length of an average fish at tl_dead = 224mm (the tl_dead for FER71).

##### Impute tl_alive from tl_dead for sample FER71 and plot linear model: `figures/impute_tl_fer71.pdf`
```{r}
# Filter only age-2 fish and retain only tl_alive, tl_dead, and river columns. Drop NA values.
full_data_age2 <- full_data %>%
  filter(consensus_age == "2") %>%
  select(tl_alive, tl_dead, river) %>% 
  drop_na()

# Run a linear mixed effect model, with stream as a random effect
model <- lme(tl_alive ~ tl_dead, 
             random = ~1|river, 
             full_data_age2)

# Summarize model output and coefficients
model_coefficients <- summary(model)

# Get r-squared value
model_r <- r.squaredGLMM(model)

# Calculate tl_alive at 224mm tl_dead using model output coefficients:
length <- 1.013964*224 + 3.900107

# Plot linear model
pdf("figures/impute_tl_fer71.pdf", width = 6, height = 5)

ggplot(full_data_age2, aes(x = tl_dead, y = tl_alive)) + 
  geom_point(fill = "grey", color = "black", pch = 21, size = 2) +
  theme_set(theme_cowplot(12)) +
  geom_smooth(method = "lm", color = "black") +
  geom_vline(xintercept = 224, linetype = "longdash", size = 0.8) + 
  geom_hline(yintercept = 231, linetype = "longdash", size = 0.8) +
  labs(x = "Total length dead (mm)", y = "Total length alive (mm)") +
  theme(axis.text = element_text(size = 15)) + 
  theme(axis.title = element_text(size = 15)) +
  scale_x_continuous(expand = c(0,0)) + 
  scale_y_continuous(expand = c(0,0)) +
  theme(panel.border = element_rect(colour = "black", fill=NA, size=1))

dev.off()
```

This figure is the basis for Supplementary Figure (XXX) in the final manuscript.

We found a strong correlation between total length dead and total length alive (<i>p</i> = < 0.0001; r<sup>2</sup> = 0.955), with the equation:

tl_alive(tl_dead) = 1.013964(tl_dead) + 3.900107

Thus, tl_alive(224) = 231.028, which we rounded to 231 mm. We input this value for tl_alive for this sample for all subsequent analyses.

##### 5b.1. Input the tl_alive value for sample FER71 (231) in the full dataframe; run the Rmd chunk below.
```{r}
# Input the calculated value for tl_alive for sample FER71
full_data$tl_alive[116] <- "231"

# Convert tl_alive column back to numeric
full_data <- full_data %>%
  mutate(tl_alive = as.numeric(tl_alive))
```

#### 5b. Create new column in full data to categorize samples by their size bin (25 mm size intervals); run the Rmd chunk below.
As part of the experimental design of this study, we aimed to collect approximately 10 fish per 25 mm size cohort, starting with 200 mm total length when alive (tl_alive = 200-224.9mm, 225-249.9mm, etc.) in order to run models for individual growth rate and maximum total length (see Analysis 4).

To summarize the number of fish captured within each size cohort, we created a new variable (column) in the dataset (size_bin), assigning each individual to 25 mm size categories.

##### Add size bin column to dataframe
```{r}
# Categorize sample ids into 25 mm size bins based on total length while alive (tl_alive)
full_data$size_bin <- with(full_data, ifelse(tl_alive>=200 & tl_alive<225, "200-225",
                                             ifelse(tl_alive >=225 & tl_alive<250, "225-250",
                                                    ifelse(tl_alive >=250 & tl_alive<275, "250-275",
                                                           ifelse(tl_alive >=275 & tl_alive<300, "275-300",
                                                                  ifelse(tl_alive >=300 & tl_alive<325, "300-325",
                                                                         ifelse(tl_alive >=325 & tl_alive<350, "325-350",
                                                                                ifelse(tl_alive >=350 & tl_alive<375, "350-375",
                                                                                       ifelse(tl_alive >=375 & tl_alive<400, "375-400",
                                                                                              ifelse(tl_alive >=400, "400+","done"))))))))))

# Convert size bin column to factor
full_data <- full_data %>%
  mutate(size_bin = factor(size_bin))

# Save the fully merged and cleaned data
save(full_data, file = "data/processed_raw_data/full_data.Rda")
```

### ----------------------- END OF PHASE 1: DATA READ-IN AND PREPARATION ----------------------- ###

## PHASE 2: DATA SUMMARIZATION AND ENUMERATION
In this phase of the analysis, we fully summarize the data so that we know how many samples belong to each group of interest for the final manuscript. 

### STEP 1: Summarize the full genotype and phenotype datasets
In this step, we are summarizing the full data to enumerate samples by groups of interest. We also ensure that both datasets contain the same samples.

#### 1b. Read in and summarize the full data; run the Rmd chunk below:

##### Summarize full genotype data
```{r}
# Load in full dataset
load("data/processed_raw_data/full_data.Rda")

# Count total number of fish analyzed
full_data %>%
  count()

# Summarize full data by range_id (population; Neosho Bass (samples) and Smallmouth Bass (reference))
full_data %>% 
  group_by(range_id) %>%
  count()

# Summarize full data by river within range (Neosho Bass, NB; Smallmouth Bass (SMB))
full_data %>% 
  group_by(range_id, river) %>%
  count()

# Summarize full data by sex
full_data %>% 
  group_by(sex) %>%
  count()

# Summarize full data by sex within species
full_data %>% 
  group_by(sex, river) %>%
  count()

# Summarize full data by size bin within river
full_data %>% 
  group_by(size_bin, river) %>%
  count()

# Summarize full data by size bin for sex within river
full_data %>% 
  group_by(size_bin, river, sex) %>%
  count()
```

### Data summary:

## Total
<b><i>N</i><sub>total</sub> = 136 </b><br>

## By species (native range)
<b><i>N</i><sub>Neosho_Bass</sub> = 116 </b><br>
<i>N</i><sub>Smallmouth Bass</sub> = 20 <br>

## By river (wihtin species)
<i>N</i><sub>nb_elk</sub> = 70 <br>
<i>N</i><sub>nb_bigsugar</sub> = 46 <br>

<i>N</i><sub>smb_white</sub> = 2 <br>
<i>N</i><sub>smb_tablerock</sub> = 8 <br>
<i>N</i><sub>smb_crooked</sub> = 10 <br>

## By sex (within Neosho Bass range)
<i>N</i><sub>male</sub> = 57 <br>
<i>N</i><sub>female</sub> = 59 <br>

## By sex within river
<i>N</i><sub>elk_male</sub> = 38 <br>
<i>N</i><sub>elk_female</sub> = 32 <br>

<i>N</i><sub>bigsugar_male</sub> = 19 <br>
<i>N</i><sub>bigsugar_female</sub> = 27 <br>

## By size bin within river
<i>N</i><sub>bigsugar_200-225</sub> = 10 <br>
<i>N</i><sub>bigsugar_225-250</sub> = 5 <br>
<i>N</i><sub>bigsugar_250-275</sub> = 3 <br>
<i>N</i><sub>bigsugar_275-300</sub> = 8 <br>
<i>N</i><sub>bigsugar_300-325</sub> = 11 <br>
<i>N</i><sub>bigsugar_325-350</sub> = 4 <br>
<i>N</i><sub>bigsugar_350-375</sub> = 1 <br>
<i>N</i><sub>bigsugar_375-400+</sub> = 4 <br>

<i>N</i><sub>elk_200-225</sub> = 10 <br>
<i>N</i><sub>elk_225-250</sub> = 14 <br>
<i>N</i><sub>elk_250-275</sub> = 17 <br>
<i>N</i><sub>elk_275-300</sub> = 12 <br>
<i>N</i><sub>elk_300-325</sub> = 7 <br>
<i>N</i><sub>elk_325-350</sub> = 3 <br>
<i>N</i><sub>elk_350-375</sub> = 1 <br>
<i>N</i><sub>elk_375-400+</sub> = 6 <br>

## By size bin within sex within river
<i>N</i><sub>bigsugar_200-225_female</sub> = 6 <br>
<i>N</i><sub>bigsugar_200-225_male</sub> = 4 <br>
<i>N</i><sub>bigsugar_225-250_female</sub> = 1 <br>
<i>N</i><sub>bigsugar_225-250_male</sub> = 4 <br>
<i>N</i><sub>bigsugar_250-275_female</sub> = 3 <br>
<i>N</i><sub>bigsugar_250-275_male</sub> = 0 <br>
<i>N</i><sub>bigsugar_275-300_female</sub> = 5 <br>
<i>N</i><sub>bigsugar_275-300_male</sub> = 3 <br>
<i>N</i><sub>bigsugar_300-325_female</sub> = 5 <br>
<i>N</i><sub>bigsugar_300-325_male</sub> = 6 <br>
<i>N</i><sub>bigsugar_325-350_female</sub> = 4 <br>
<i>N</i><sub>bigsugar_325-350_male</sub> = 0 <br>
<i>N</i><sub>bigsugar_350-375_female</sub> = 0 <br>
<i>N</i><sub>bigsugar_350-375_male</sub> = 1 <br>
<i>N</i><sub>bigsugar_375-400+_female</sub> = 3 <br>
<i>N</i><sub>bigsugar_375-400+_male</sub> = 1 <br>

<i>N</i><sub>elk_200-225_female</sub> = 4 <br>
<i>N</i><sub>elk_200-225_male</sub> = 6 <br>
<i>N</i><sub>elk_225-250_female</sub> = 8 <br>
<i>N</i><sub>elk_225-250_male</sub> = 6 <br>
<i>N</i><sub>elk_250-275_female</sub> = 8 <br>
<i>N</i><sub>elk_250-275_male</sub> = 9 <br>
<i>N</i><sub>elk_275-300_female</sub> = 5 <br>
<i>N</i><sub>elk_275-300_male</sub> = 7 <br>
<i>N</i><sub>elk_300-325_female</sub> = 2 <br>
<i>N</i><sub>elk_300-325_male</sub> = 5 <br>
<i>N</i><sub>elk_325-350_female</sub> = 2 <br>
<i>N</i><sub>elk_325-350_male</sub> = 1 <br>
<i>N</i><sub>elk_350-375_female</sub> = 1 <br>
<i>N</i><sub>elk_350-375_male</sub> = 0 <br>
<i>N</i><sub>elk_375-400+_female</sub> = 2 <br>
<i>N</i><sub>elk_375-400+_male</sub> = 4 <br>

These sample sizes were used as the basis for Table 1a and the sample size (N) column of Table S1 in the final manuscript.

### ----------------------- END OF PHASE 2: DATA SUMMARIZATION AND ENUMERATION ----------------------- ###

### ----------------------- END OF ANALYSIS 2: FILTERING ANALYSIS ----------------------- ###


