---
title: "Analysis 1: Filtering Analysis"
author: "Joe Gunn"
date: "2022-07-28"
output: html_document
---

# Project: Effects of admixture on fitness in Neosho Bass populations 
<font size="+1">We assessed the effect of admixture on fitness in two stream populations within the native range of the Neosho Bass (<i>M. velox</i>) which are known to have extensively hybridized with Smallmouth Bass (<i>Micropterus dolomieu</i>). Specifically, we used 14 microsatellite loci in a Bayesian analysis of population structure to estimate proportions of interspecific ancestry in individuals collected from Big Sugar Creek and the Elk River in southwestern Missouri (Central Interior Highlands ecoregion (CIH), North America). We used ancestry inference to identify fish as "Pure Neosho Bass", "Pure Smallmouth Bass", or "Admixed". For each group, we measured age and total length and projected individual growth using the standard parameterization of the von Berlanffy growth model, comparing average theoretical maximum length among groups. Finally, we used calculated a body condition as a proxy of fitness and generated heterozygosity-fitness correlations of body condition across the global dataset, within stream populations, and within ancestry groups. We ultimately sought to understand the short-term genetic consequences of admixture for Neosho Bass populations in order to better inform management and long-term viability of distinct, economically and ecologically important sportfish species in the CIH.</font>

## Specific Aim: Data preparation, enumeration, and filtering
For this aim, we clean, filter, and summarize the full genotype data (14 microsatellite loci) and phenotypic data (consensus age and total length) for our raw samples, which were fish collected from two streams (Big Sugar Creek and Elk River) within the recognized native range of the Neosho Bass (<i>M. velox</i>). We also clean and filter the full genotype data for a reference set of samples representing Smallmouth Bass (<i>M. dolomieu</i>) collected from streams in the White River drainage in the Smallmouth Bass native range. We did not collect phenotypic data on the Smallmouth Bass reference samples.

## Phases of Analysis
### Phase 1: Data read-in and preparation
### Phase 2: Data filtering and enumeration

## Libraries needed for analysis
```{r setup, include=FALSE, include = FALSE}
library(readxl)
library(tidyverse)
library(cowplot)
library(devtools)
library(pophelper)
library(genepopedit)
```

## PHASE 1: DATA READ-IN AND PREPARATION

### STEP 1: Read in metadata for samples (all fish collected from the Neosho Bass range) and references (all fish collected from the Smallmouth Bass range)
In this step, we read in previously curated metadata files for the sample set of fish collected from two streams within the Neosho Bass native range (`../raw_data/sample_metadata.xlsx`) and for the reference set of fish collected from three streams within the Smallmouth Bass native range (`../raw_data/reference_metadata.xlsx`). We manipulate and clean the dataset, and enumerate samples in STEP 4 below.

#### 1a. Read in and clean sample metadata; run the Rmd chunk below.

##### Read in and clean sample metadata:
```{r}
# Read in sample metadata
sample_metadata <- read_excel("../raw_data/sample_metadata.xlsx") 

# Convert characters to factors
sample_metadata <- sample_metadata %>%
  mutate(sample_id = factor(sample_id)) %>%
  mutate(river_code = factor(river_code)) %>%
  mutate(sample_number = as.numeric(sample_number)) %>%
  mutate(easting = as.numeric(easting)) %>%
  mutate(northing = as.numeric(northing)) %>%
  mutate(range_id = factor(range_id)) %>%
  mutate(river = factor(river)) %>%
  mutate(population = factor(population))
```

#### 1b. Read in and clean reference metadata; run the Rmd chunk below.

##### Read in and clean reference metadata:
```{r}
# Read in reference metadata
reference_metadata <- read_excel("../raw_data/reference_metadata.xlsx") 

# Convert characters to factors
reference_metadata <- reference_metadata %>%
  mutate(sample_id = factor(sample_id)) %>%
  mutate(river_code = factor(river_code)) %>%
  mutate(sample_number = as.numeric(sample_number)) %>%
  mutate(easting = as.numeric(easting)) %>%
  mutate(northing = as.numeric(northing)) %>%
  mutate(range_id = factor(range_id)) %>%
  mutate(river = factor(river)) %>%
  mutate(population = factor(population))
```

#### 1c. Concatenate sample and reference metadata; run the Rmd chunk below.

##### Read in and clean reference metadata:
```{r}
# Rbind sample and reference metadata
metadata <- rbind(sample_metadata, reference_metadata)

# Save metadata in processed raw data directory in the filtering analysis
save(metadata, file = "data/processed_raw_data/metadata.Rda")
```

### STEP 2: Read in genotype data for all samples and references (one single genotype file)
In this step, we read in genotype data for all fish included in the study, including samples (collected from the Neosho Bass native range) and references (collected from the Smallmouth Bass native range). These genotype data were derived from multiplex PCR (see `../project_info/msat_primer_info/msat_primer_info.docx` for microsatellite primer oligonucleotide flanking sequences and corresponding metadata, including literature references) followed by and fragment analysis and subsequent allele scoring in GeneMarker (Soft Genetics). We manipulate and clean the dataset, and enumerate samples in STEP 4 below.

#### 2a. Read in, clean, and join genotype data with metadata; run the Rmd chunk below.

##### Read in and clean genotype data:
```{r}
# Read in genotype data
genotype_data <- read_excel("../raw_data/genotype_data.xlsx") 

# Merge metadata and genotype data
genotype_data <- merge(metadata, genotype_data, by = "sample_id")

# Save combined metadata and genotype data in processed raw data directory in the filtering analysis
save(genotype_data, file = "data/processed_raw_data/metadata_genotype_data.Rda")
```

### STEP 3: Read in phenotype data data for samples
In this step, we read in phenotype data for all fish included in the study. Data include: "sample_id" (same as in metadata and genotype data), "sex" (male or female, determined by internal gonad examination), "tl_alive" (total length of the fish immediately upon capture by rod-and-reel, as measured from the tip of the lower mandible to the posterior tip of the caudal fin squeezed together), "tl_dead" (total length of the fish upon sample processesing (after approximately one month frozen)), "mass_dead" (mass of the fish in grams at time of sample processing), "meas_ints" (initials of the person doing the phenotypic measuring, measuring, or assessment, for later analysis of observer/measurement bias), "eddie_age" (age estimate based on saggital otolith annuli, performed by Eddie Sterling), "joe_age" (age estimate performed by Joe Gunn), "michael_age" (age estimate performed by Michael Moore), "consensus_age" (ultimate age estimate based on consensus of three agers, Eddie, Joe, and Michael), "sl" (standard length of the fish, as measured from the tip of the mandible to the hyperal plate, just anterior to the caudal fin), "bd" (body depth, as measured at the thickest part of the fish), "hl" (head length, as measured from the tip of the lower mandible to the posterior end of the gill plate), "ol" (orbital length, the diameter of the eye ball), and "sd_ray" (the number of soft dorsal fin rays). We manipulate and clean the dataset, and enumerate samples in STEP 4 below.

#### 2a. Read in, clean, and join genotype data with metadata; run the Rmd chunk below.

##### Read in and clean genotype data:
```{r}
# Read in genotype data
phenotype_data <- read_excel("../raw_data/phenotype_data.xlsx") 

```